game/missiongenerator/luagenerator.py

```diff
@@ -210,6 +211,49 @@ def generate_plugin_data(self) -> None:
                     iads_element.add_key_value(property, value)
             for role, connections in node.connections.items():
                 iads_element.add_data_array(role, connections)
+        
+        # Add artillery and support units info
+        artillery_object = lua_data.add_item("artilleryGroups")
+        ground_artillery_group_collection = artillery_object.get_or_create_item("groundArtillery")
+        ship_artillery_group_collection = artillery_object.get_or_create_item("shipArtillery")
+
+        # First add all artillery units that are theater objects (mostly ships)
+        for cp in self.game.theater.control_points_for(player=True):
+            for ground_object in cp.ground_objects:
+                for group in ground_object.groups:
+                    # Check if first unit in group is ground-based or ship artillery
+                    unit = group.units[0]
+                    if unit.unit_type is None:
+                        continue
+                    if unit.unit_type.unit_class == UnitClass.ARTILLERY:
+                        ground_artillery_group = ground_artillery_group_collection.add_item()
+                        ground_artillery_group.add_key_value("groupName", group.group_name)
+                    elif unit.unit_type.unit_class in (UnitClass.CRUISER, UnitClass.DESTROYER, UnitClass.FRIGATE):
+                        # TODO: we assume that these ship classes have guns... Which might not be the case.
+                        ship_artillery_group = ship_artillery_group_collection.add_item()
+                        ship_artillery_group.add_key_value("groupName", group.group_name)
+        
+        # Add artillery that are frontline groups
+        for frontline_group in self.mission_data.player_frontline_groups:
+            if frontline_group.unit_type.unit_class is None:
+                continue
+            if frontline_group.unit_type.unit_class == UnitClass.ARTILLERY:
+                ground_artillery_group = ground_artillery_group_collection.add_item()
+                ground_artillery_group.add_key_value("groupName", frontline_group.group_name)
+            
+
+        # Add forward observer (FO) (TODO: maybe adding new flight type "Foward Observer"?)
+        forward_observer_object = lua_data.add_item("forwardObserverUnits")
+        for flight in self.mission_data.flights:
+            if not flight.friendly or not len(flight.client_units) > 0:
+                continue
+            if flight.flight_type == FlightType.ARMED_RECON:
+                for unit in flight.client_units:
```

Comment:
I wasn't sure if that was necessary since it is using the `client_units` property, but I'm absolutely not familiar with the codebase, so I've also added a check for player or client skill, thanks! 😄

Reply:
Right... I got confused there as well, the check on skill-level can be skipped because that's exactly what `client_units` will do down the line... my bad 😅